File: content/problems/2550-count-collisions-of-monkeys-on-a-polygon/solutions/modular-power-minus-two/Solution.kt
class Solution {
    fun monkeyMove(nInput:Int):Int{
        val mod=1_000_000_007L;
        var n=nInput;
        var base=2L;
        var result=1L;
        while(n>0){
            if(n and 1==1)result=result*base%mod;
            base=base*base%mod;
            n=n shr 1
        };
        return ((result-2+mod)%mod).toInt()
    }
}

File: content/problems/2550-count-collisions-of-monkeys-on-a-polygon/solutions/modular-power-minus-two/solution.cpp
class Solution {
    public:int monkeyMove(int n){
        const long long mod=1'000'000'007;
        long long base=2,result=1;
        while(n){
            if(n&1)result=result*base%mod;
            base=base*base%mod;
            n>>=1;
        }
        return (result-2+mod)%mod;
    }
};
